provider-example/src/hpke.rs

```diff
@@ -5,77 +5,124 @@ use core::fmt::{Debug, Formatter};
+        kdf_id: HpkeKdfId::HKDF_SHA256,
+        aead_id: HpkeAeadId::AES_128_GCM,
+    },
+});
+
+pub static DHKEM_X25519_HKDF_SHA256_AES_256: &HpkeRs = &HpkeRs(HpkeSuite {
+    kem: HpkeKemId::DHKEM_X25519_HKDF_SHA256,
+    sym: HpkeSymmetricCipherSuite {
+        kdf_id: HpkeKdfId::HKDF_SHA256,
+        aead_id: HpkeAeadId::AES_256_GCM,
+    },
+});
+
+pub static DHKEM_X25519_HKDF_SHA256_CHACHA20_POLY1305: &HpkeRs = &HpkeRs(HpkeSuite {
+    kem: HpkeKemId::DHKEM_X25519_HKDF_SHA256,
+    sym: HpkeSymmetricCipherSuite {
+        kdf_id: HpkeKdfId::HKDF_SHA256,
+        aead_id: HpkeAeadId::CHACHA20_POLY_1305,
+    },
+});
+
+/// A HPKE suite backed by the [hpke-rs] crate and its rust-crypto cryptography provider.
+///
+/// [hpke-rs]: https://github.com/franziskuskiefer/hpke-rs
+pub struct HpkeRs(HpkeSuite);
+
+impl HpkeRs {
+    fn start(&self) -> Result<hpke_rs::Hpke<HpkeRustCrypto>, Error> {
+        Ok(hpke_rs::Hpke::new(
             hpke_rs::Mode::Base,
-            KemAlgorithm::try_from(u16::from(suite.kem)).map_err(other_err)?,
-            KdfAlgorithm::try_from(u16::from(suite.sym.kdf_id)).map_err(other_err)?,
-            AeadAlgorithm::try_from(u16::from(suite.sym.aead_id)).map_err(other_err)?,
-        ))))
-    }
-
-    fn supports_suite(&self, suite: &HpkeSuite) -> bool {
-        let kem = KemAlgorithm::try_from(u16::from(suite.kem)).ok();
-        let kdf = KdfAlgorithm::try_from(u16::from(suite.sym.kdf_id)).ok();
-        let aead = AeadAlgorithm::try_from(u16::from(suite.sym.aead_id)).ok();
-        match (kem, kdf, aead) {
-            (Some(kem), Some(kdf), Some(aead)) => {
-                HpkeRustCrypto::supports_kem(kem).is_ok()
-                    && HpkeRustCrypto::supports_kdf(kdf).is_ok()
-                    && HpkeRustCrypto::supports_aead(aead).is_ok()
-            }
-            _ => false,
-        }
+            KemAlgorithm::try_from(u16::from(self.0.kem)).map_err(other_err)?,
+            KdfAlgorithm::try_from(u16::from(self.0.sym.kdf_id)).map_err(other_err)?,
+            AeadAlgorithm::try_from(u16::from(self.0.sym.aead_id)).map_err(other_err)?,
+        ))
     }
 }
 
-struct HpkeRs(hpke_rs::Hpke<HpkeRustCrypto>);
-
 impl Debug for HpkeRs {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("HpkeRsHpke").finish()
+        f.debug_struct("HpkeRs")
+            .field("suite", &self.0)
+            .finish()
     }
 }
 
 impl Hpke for HpkeRs {
     fn seal(
-        &mut self,
+        &self,
         info: &[u8],
         aad: &[u8],
         plaintext: &[u8],
         pub_key: &HpkePublicKey,
     ) -> Result<(EncapsulatedSecret, Vec<u8>), Error> {
         let pk_r = hpke_rs::HpkePublicKey::new(pub_key.0.clone());
         let (enc, ciphertext) = self
-            .0
+            .start()?
             .seal(&pk_r, info, aad, plaintext, None, None, None)
             .map_err(other_err)?;
         Ok((EncapsulatedSecret(enc.to_vec()), ciphertext))
```

Comment:
When using AES GCM, can the `nonce` value be included in the `EncapsulatedSecret` (or separately returned) as the same value is required when opening